lib/arux_app/api.rb

```diff
@@ -1,29 +1,27 @@
 module AruxApp
   module API
-    class << self
-      @@mode = :standard
-      [:test, :dev, :standard].each do |m|
-        define_method("#{m}mode?") do
-          @@mode == m
-        end
+    DOMAINS = {
+      production: "arux.app",
+      staging: "arux.blue",
+      development: HOSTNAME,
+      test: "arux.test"
+    }
 
-        define_method("#{m}mode") do
-          @@mode == m
-        end
+    class << self
+      def mode
+        raise "Environment is not set, i.e. ARUX_APP_GEM_MODE = :development" unless const_defined?(:ARUX_APP_GEM_MODE)
```

Comment:
Any reason not to do the same check as in `domain`? And if we do it here, I think we could remove line 23 too

```suggestion
        raise "Environment is not set to a valid environment, i.e. ARUX_APP_GEM_MODE = :development" if !DOMAINS.has_key?(ARUX_APP_GEM_MODE)
```

Reply:
I played around with that, and it appears we need the const_defined method while passing in a symbol. Otherwise we get something like this
```
        5: from /Users/brooksgeenen/arux/arux_app_gem/lib/arux_app/api.rb:12:in `mode'
        4: from /Users/brooksgeenen/.rvm/gems/ruby-2.5.0-nrbgem18@tce-v2/gems/activesupport-2.3.18/lib/active_support/dependencies.rb:120:in `const_missing'
        3: from /Users/brooksgeenen/.rvm/gems/ruby-2.5.0-nrbgem18@tce-v2/gems/activesupport-2.3.18/lib/active_support/dependencies.rb:130:in `rescue in const_missing'
        2: from /Users/brooksgeenen/.rvm/gems/ruby-2.5.0-nrbgem18@tce-v2/gems/activesupport-2.3.18/lib/active_support/dependencies.rb:417:in `qualified_name_for'
        1: from /Users/brooksgeenen/.rvm/gems/ruby-2.5.0-nrbgem18@tce-v2/gems/activesupport-2.3.18/lib/active_support/dependencies.rb:611:in `to_constant_name'
ArgumentError (Anonymous modules have no name to be referenced by)
```

Reply:
`const_get(:ARUX_APP_GEM)` maybe?

Or check for it to be defined, but still check if it's a valid key in `DOMAINS`?

Or leave it as-is; what's here is fine too.

Reply:
> `const_get(:ARUX_APP_GEM)` maybe?

Hmm, still throws the same error, I think I might leave as is for now, since they are both working in tandem and checking each other.